seaworthy/tests-core/test_logs.py

```diff
@@ -255,12 +252,76 @@ def test_streaming_waits(self):
         t0 = datetime.now()
         self.assertEqual(self.stream(con), [b'hello\n', b'goodbye\n'])
         t1 = datetime.now()
-        self.assertEqual(self.stream(con), [b'hello\n', b'goodbye\n'])
+        self.assertEqual(self.stream(con), [])
         t2 = datetime.now()
         self.assertLess(0.3, (t1 - t0).total_seconds())
         self.assertLess((t2 - t1).total_seconds(), 0.3)
 
 
+class TestStreamWithHistoryFunc(unittest.TestCase):
+    def mkcontainer(self, *args, **kw):
+        con = FakeLogsContainer(*args, **kw)
+        self.addCleanup(con.cleanup)
+        return con
+
+    def stream(self, con, timeout=1):
+        # Always clean up the feeder machinery when we're done. This is only
+        # necessary if we timed out, but it doesn't hurt to clean up an
+        # already-clean FakeLogsContainer.
+        try:
+            return list(stream_logs(con, timeout=timeout))
+        finally:
+            con.cancel_feeder()
```

Comment:
Why not `con.cleanup()`?

Reply:
Hysterical raisins. I've updated all the places.